core/src/test/java/io/aiven/klaw/service/TopicControllerServiceTest.java

```diff
@@ -382,74 +685,219 @@ public void getAllTopics() {
+        .thenReturn(List.of(getTopic("testtopic")));
+    when(commonUtilsService.getFilteredTopicsForTenant(any())).thenReturn(utilMethods.getTopics());
+    when(commonUtilsService.getTeamId(anyString())).thenReturn(1);
+
+    when(handleDbRequests.updateTopicDocumentation(any()))
+        .thenReturn(ApiResultStatus.FAILURE.value);
+
+    ApiResponse apiResponse = topicControllerService.saveTopicDocumentation(topicInfo);
+    assertThat(apiResponse.getResult()).isEqualTo(ApiResultStatus.FAILURE.value);
+  }
+
+  @Test
+  @Order(32)
+  public void getTopicDetailsPerEnvFailure1() {
+    stubUserInfo();
+    String envId = "1", topicName = "testtopic";
+    when(commonUtilsService.getTenantId(anyString())).thenReturn(101);
+    when(handleDbRequests.getTopics(anyString(), anyInt())).thenReturn(Collections.emptyList());
+    when(commonUtilsService.getEnvsFromUserId(anyString()))
+        .thenReturn(new HashSet<>(Collections.singletonList("1")));
+
+    Map<String, Object> topicDetailsPerEnvResponse =
+        topicControllerService.getTopicDetailsPerEnv(envId, topicName);
+    assertThat(topicDetailsPerEnvResponse.get("error")).isEqualTo("Topic does not exist.");
+  }
+
+  @Test
+  @Order(33)
+  public void getTopicDetailsPerEnvFailure2() {
+    stubUserInfo();
+    String envId = "1", topicName = "testtopic";
+    when(commonUtilsService.getTenantId(anyString())).thenReturn(101);
+    when(handleDbRequests.getTopics(anyString(), anyInt())).thenReturn(utilMethods.getTopics());
+    when(commonUtilsService.getEnvsFromUserId(anyString()))
+        .thenReturn(new HashSet<>(Collections.singletonList("1")));
+    when(manageDatabase.getKafkaEnvList(anyInt())).thenReturn(utilMethods.getEnvLists());
+
+    Map<String, Object> topicDetailsPerEnvResponse =
+        topicControllerService.getTopicDetailsPerEnv(envId, topicName);
+    assertThat(topicDetailsPerEnvResponse.get("error"))
+        .isEqualTo("Sorry, your team does not own the topic !!");
+  }
+
+  @Test
+  @Order(34)
+  public void getTopicDetailsPerEnvSuccess() {
+    stubUserInfo();
+    String envId = "1", topicName = "testtopic";
+    when(commonUtilsService.getTenantId(anyString())).thenReturn(101);
+    when(handleDbRequests.getTopics(anyString(), anyInt())).thenReturn(utilMethods.getTopics());
+    when(commonUtilsService.getEnvsFromUserId(anyString()))
+        .thenReturn(new HashSet<>(Collections.singletonList("1")));
+    when(manageDatabase.getKafkaEnvList(anyInt())).thenReturn(utilMethods.getEnvLists());
+    when(commonUtilsService.getTeamId(anyString())).thenReturn(3);
+
+    Map<String, Object> topicDetailsPerEnvResponse =
+        topicControllerService.getTopicDetailsPerEnv(envId, topicName);
+    assertThat(topicDetailsPerEnvResponse.get("topicExists")).isEqualTo(true);
+    assertThat(topicDetailsPerEnvResponse.get("topicContents")).isNotNull();
+  }
+
+  @Test
+  @Order(35)
+  public void getTopicsSuccess1() {
```

Comment:
We should give it a proper name instead of `1` and `2`.

Reply:
Agree, updated.